Reorganize Classic UI Theming part

stevepiercy opened the pull request:
Closes #1645

<!-- readthedocs-preview plone6 start -->
----
📚 Documentation preview 📚: https://plone6--1831.org.readthedocs.build/classic-ui/theming/index.html

<!-- readthedocs-preview plone6 end -->

stevepiercy committed:
Reorganize Classic UI Theming part

stevepiercy commented:
@plone/classicui-team please review.

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy commented:
First person to review this PR gets a 🍪 !

@petschki
@yurj
@MrTango
@jensens
@1letter

It has one section where I need some help to clarify it, please!

petschki commented on docs/classic-ui/theming/settings-ttw.md:
```diff
@@ -0,0 +1,62 @@
+
+This chapter describes how to change Classic UI theme settings through-the-web ({term}`TTW`) in Plone 6.
+
+Small theme changes can be made via control panels or by updating Plone 6 Classic UI's `custom.css`.
```
```suggestion
Small theme changes can be made via control panels.
```

There is no `custom.css` which you could update.

stevepiercy replied:
There is a `custom.css` generated when you go through the process under [Custom Styles](https://plone6--1831.org.readthedocs.build/classic-ui/theming/settings-ttw.html#custom-styles). But, yeah, I see your point. It's an implementation detail that most users won't even know, unless they view the rendered HTML. Plus a Site Admin does it through a control panel anyway.

petschki replied:
Ah... I know what you mean now. It would be event more problematic if a user looks at the source code and tries to find the `custom.css` file anywhere, but this doesn't physically exists anywhere because its generated. This should perhaps be explained in the chapter "custom styles"

stevepiercy committed:
Update docs/classic-ui/theming/settings-ttw.md

Co-authored-by: Peter Mathis <petschki@users.noreply.github.com>

petschki commented on docs/classic-ui/theming/settings-ttw.md:
```diff
@@ -0,0 +1,62 @@
+Navigate to {menuselection}`Admin --> Site Setup --> Theming --> Advanced settings --> Custom Styles`, or visit the URL path `/@@theming-controlpanel#autotoc-item-autotoc-2` in your web browser's address bar.
+
+Enter any arbitrary styles, and save your changes.
+The changes are stored in the file {file}`custom.css`.
```
This is not really a file but a BrowserView called `custom.css` ... might be missleading to some users.

stevepiercy replied:
Yeah, it misled me. 🙃 Fixed in https://github.com/plone/documentation/pull/1831/commits/08cda10a73700ea6949f091d68f94e8e9d03dc18

stevepiercy commented:
We have to ignore the pull request preview build failure for a while. I just filed an issue with RTD at https://github.com/readthedocs/readthedocs.org/issues/11912

stevepiercy committed:
Update settings-ttw.md

petschki reviewed (dismissed):
I'm happy with the restructuring here. Thank you!

petschki commented:
Where's my 🍪 😉

stevepiercy commented:
> Where's my 🍪 😉

I'll bring a batch for the next PloneConf or sprint. 😁 

I need help with this TODO, as I don't have the experience or knowledge. If y'all can give me some notes, I can wordsmith something together.

https://plone6--1831.org.readthedocs.build/classic-ui/theming/index.html

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy commented:
@petschki @yurj @MrTango @jensens @1letter

I need help with this TODO about choosing a theming method, as I don't have the experience or knowledge. If y'all can give me some notes or guidance, I can wordsmith something together and finish up this PR.

https://plone6--1831.org.readthedocs.build/classic-ui/theming/index.html

Thank you!

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

ale-rt commented:
@stevepiercy what about merging master and merging this PR with the TODO?
We can add it later.

stevepiercy commented:
@ale-rt we've tried that, and it doesn't work. A little bit of evidence:

- https://6.docs.plone.org/classic-ui/forms.html
- https://6.docs.plone.org/classic-ui/templates.html
- https://6.docs.plone.org/backend/annotations.html
- https://6.docs.plone.org/backend/configuration-registry.html
- https://6.docs.plone.org/backend/search.html
- https://6.docs.plone.org/backend/security.html
- https://6.docs.plone.org/backend/sending-email.html
- https://6.docs.plone.org/backend/subscribers.html
- https://6.docs.plone.org/backend/users-groups.html
- https://6.docs.plone.org/backend/vocabularies.html
- https://6.docs.plone.org/backend/workflows.html
- https://6.docs.plone.org/backend/zodb.html

There's also the embarrassment of completing only [13% of Plone 6.0 and 6.1 documentation issues](https://github.com/plone/documentation/milestone/9) at the time of Plone 6.1 release.

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy commented:
> @petschki @yurj @MrTango @jensens @1letter
> 
> I need help with this TODO about choosing a theming method, as I don't have the experience or knowledge. If y'all can give me some notes or guidance, I can wordsmith something together and finish up this PR.
> 
> [](https://plone6--1831.org.readthedocs.build/classic-ui/theming/index.html)

Since I got no response, I'm inclined to refer readers to get help to make a choice in the Classic UI Community Forum topic and Discord channel.

I'll revisit this when I get some time, but it can't be merged until helpful content replaces the todo.

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Fix broken link

jensens commented:
I  think that TTW is likely for testing or small-scale changes, especially for short-lived or prototype sites. It allows you to make minor adjustments without needing to dive into code or production-level setups. TTW can be convenient for quick experiments, for websites that don't require complex theming or short-lived websites.

Add-ons are definitely suited for production environments. An add-on allows you to keep your customization modular, and it provides a more structured approach to larger-scale theming. This method is ideal for more permanent or complex customizations since it enables easier maintenance and portability.

The customized Barceloneta theme, using Diazo, and creating a theme from scratch offer even more flexibility:

-  Barceloneta theme: Ideal for a starting point, especially if you want to tweak it as an existing, solid theme. You can customize its components and styles without starting from zero.

-  From Scratch: If your project requires something completely unique and none of the existing themes fit, building a theme from scratch would be the way to go. This is most appropriate for highly customized websites with distinct design requirements.

-  Diazo: Additional to an addon with customized or from-scratch and one of the more advanced methods. It  offers control over the markup without customizing templates. It's great to have for more complex themes where content needs to remain flexible but you want full control over the markup. Disadvantage is possible slow down of the rendering process and more complex and more difficult to maintain theme. 

In summary:

-   Use TTW for minor changes or testing.
-   Use Add-ons for production-level theming that needs scalability, maintainability, and modularity to customize Barceloneta or write from-scratch themes.
-   Use Diazo in your addon if you need to modify the markup without touching the page-templates.

stevepiercy committed:
Merge branch '6.0' into classic-ui-theming-reorg-1645

stevepiercy committed:
Add guide for choosing one Classic UI theme or another, thanks to @jensens!

stevepiercy commented:
@jensens that is beautiful. Thank you so much! I wordsmithed it into the index page in 95aeba3ab47e18142b6dfdb9bb5501c66fdb7cb5.

With that, I approve this PR, pending the merge of the related PR https://github.com/plone/documentation/pull/1870 from @erral.

erral committed:
Provide the content for the Diazo theming part (#1870)

* Update tips submodules/plone.api submodules/volto

* Update tip submodules/volto

* Fix broken references (#1868)

* Fix broken references
- `link.svg`
- `_inc/_install-browser-reqs-volto.md`

* Fix linkcheck to bobtemplates

* Update tip submodules/volto

* Fix link

* Update tip submodules/plone.api

* Update tip submodules/volto

* initial structure of Theming with Diazo

* more doc

* initial structure of Theming with Diazo

* more doc

* - Add diazo to toctree
- Add note update getting help for choosing a theming method

* Clean up grammar, one sentence per line, MyST syntax

* Add language

* Move concepts and advice into a new Theme development advice section

* Minor tweaks

* Rename section to reflect its content

* Adjust heading level

* Update diazo.md

explain the purpose of each file created when addin a theme

* Update diazo.md

Link to Diazo docs.

* Update diazo.md

* - Make first mention of Diazo a term, already existing in the Glossary
- Reuse a sentence from Diazo docs that fills a critical gap in the introduction.
- Convert list of files descriptions into a definition list

* Tweak markup

* add tinymce templates docs

* Tidy up

* Add a label and important admonition about Template plugin removal in TinyMCE 7.0

* Remove important admonition about Template plugin

---------

Co-authored-by: Steve Piercy <web@stevepiercy.com>

stevepiercy commented:
Thank you @petschki, @jensens, and @erral for helping on this PR. Will merge at long last when it is all green.

stevepiercy committed:
Fix broken link
